ui-speedspacechart/src/components/helpers/drawElements/curve.ts

```diff
@@ -42,27 +41,39 @@ export const drawCurve = ({ ctx, width, height, store }: DrawFunctionParams) =>
   const maxSpeed = maxSpeedValue(store);
   const maxPosition = maxPositionValue(store);
 
-  ctx.lineWidth = 0.5;
-  ctx.fillStyle = 'rgba(17, 101, 180, 0.02)';
-  ctx.strokeStyle = 'rgb(17, 101, 180, 0.5)';
-
+  // Curves must be drawn twice one for the fill and one for the stroke
```

Comment:
You can draw the curve only once, if you do the things in this order:
- Describe curve path (with `moveTo` and `lineTo`)
- Draw the curve (with `stroke`)
- Close the path (with the last `lineTo` calls and `closePath`)
- Fill the curve (with `fill`)

You can see an example here:
https://codesandbox.io/p/sandbox/d367fh?file=%2Fsrc%2Findex.mjs

Reply:
if you fill after stroke, the area filled will partially overlap the stroke, right?

Reply:
Indeed. It doesn't seem possible to do this in one pass.

> [!NOTE]
> I did a little optimization. Now, x/y points positions are computed only once and reuse for the stroke.